Improve runtime tests & jenkins

- [ ] Error out on all scripts
- [ ] Jenkins must fail on test failures

Maybe also better error messages and abort the tests once a failure was found. For example:
```
# ./01-ct.sh 
+ TEST_NET=cilium
+ trap cleanup EXIT
+ cleanup
+ docker rm -f server client httpd1 httpd2 curl
+ true
+ monitor_stop
+ '[' '!' -z '' ']'
+ monitor_start
+ MONITOR_PID=3312
+ docker network inspect cilium
+ sudo cilium monitor
[]
+ docker network create --ipv6 --subnet ::1/112 --ipam-driver cilium --driver cilium cilium
panic: Unable to get object: no such file or directory

goroutine 1 [running]:
panic(0x13ab960, 0xc4213480a0)
	/usr/lib/golang/src/runtime/panic.go:500 +0x1a1
github.com/cilium/cilium/cilium/cmd.runMonitor()
	/usr/lib/golang/src/github.com/cilium/cilium/cilium/cmd/monitor.go:98 +0x3e4
github.com/cilium/cilium/cilium/cmd.glob..func14(0x1fee080, 0x2596ac0, 0x0, 0x0)
	/usr/lib/golang/src/github.com/cilium/cilium/cilium/cmd/monitor.go:41 +0x14
github.com/cilium/cilium/vendor/github.com/spf13/cobra.(*Command).execute(0x1fee080, 0x2596ac0, 0x0, 0x0, 0x1fee080, 0x2596ac0)
	/usr/lib/golang/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:648 +0x443
github.com/cilium/cilium/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x1feb820, 0xc42006c058, 0x0, 0x46779e)
	/usr/lib/golang/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:735 +0x367
github.com/cilium/cilium/vendor/github.com/spf13/cobra.(*Command).Execute(0x1feb820, 0x0, 0x0)
	/usr/lib/golang/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:693 +0x2b
github.com/cilium/cilium/cilium/cmd.Execute()
	/usr/lib/golang/src/github.com/cilium/cilium/cilium/cmd/root.go:69 +0x31
main.main()
	/usr/lib/golang/src/github.com/cilium/cilium/cilium/main.go:20 +0x14
Error response from daemon: plugin not found
+ docker run -dt --net=cilium --name server -l id.server tgraf/netperf
Unable to find image 'tgraf/netperf:latest' locally
Trying to pull repository docker.io/tgraf/netperf ... 
sha256:c16ea0875be7ee33b9e3fa3645d55336200e3cd0c6bbb1b16c7595782660ed37: Pulling from docker.io/tgraf/netperf
985c5f84712b: Pull complete 
548b4d55a521: Pull complete 
[...]
```

--- a/tests/05-cni.sh
+++ b/tests/05-cni.sh
@@ -95,7 +95,8 @@ export NETCONFPATH=`pwd`/net.d
 
 git clone 'https://github.com/containernetworking/cni'
 cd cni
-./build
+git checkout tags/v0.5.2
+./build.sh
 export CNI_PATH=`pwd`/bin
 cp "${dir}/../plugins/cilium-cni/cilium-cni" "${CNI_PATH}"
 cd scripts

--- a/tests/06-lb.sh
+++ b/tests/06-lb.sh
@@ -374,7 +374,7 @@ cilium bpf ct list $SERVER1_ID
 cilium service update --rev --frontend "$SVC_IP4:0"  --id 223 \
 			--backends "$SERVER1_IP4:0"
 
-docker exec -ti server1 ping -c 4 $SVC_IP4 || {
+docker exec -i server1 ping -c 4 $SVC_IP4 || {
 	abort "Error: Unable to reach own service IP"
 }
 

--- a/tests/06-lb.sh
+++ b/tests/06-lb.sh
@@ -234,7 +234,7 @@ docker run -dt --net=$TEST_NET --name server2 -l id.server -l server2 httpd
 docker run -dt --net=$TEST_NET --name server3 -l id.server -l server3 httpd
 docker run -dt --net=$TEST_NET --name server4 -l id.server -l server4 httpd
 docker run -dt --net=$TEST_NET --name server5 -l id.server -l server5 httpd
-docker run -dt --net=$TEST_NET --name client -l id.client noironetworks/nettools
+docker run -dt --net=$TEST_NET --name client -l id.client tgraf/nettools
 
 # FIXME IPv6 DAD period
 sleep 5

--- a/tests/06-lb.sh
+++ b/tests/06-lb.sh
@@ -379,6 +379,12 @@ docker exec -i server1 ping -c 4 $SVC_IP4 || {
 }
 
 ## Test 5: Run wrk & ab from container => bpf_lxc (LB) => local container
+# Only run these tests if BENCHMARK=1 has been set
+if [ -z $BENCHMARK ]; then
+	echo "Skipping Test 5, not in benchmark mode."
+	echo "Run with BENCHMARK=1 to enable this test"
+	exit 0
+fi
 
 cilium service update --rev --frontend "[$SVC_IP6]:80" --id 2223 \
                         --backends "[$SERVER1_IP]:80" \

--- a/tests/06-lb.sh
+++ b/tests/06-lb.sh
@@ -354,7 +354,7 @@ docker exec -i client ping6 -c 4 $SVC_IP6 || {
 }
 
 docker exec -i client ping -c 4 $SVC_IP4 || {
-	abort "Error: Unable to reach netperf TCP IPv6 endpoint"
+	abort "Error: Unable to reach netperf TCP IPv4 endpoint"
 }
 
 cilium endpoint config $CLIENT_ID Policy=false

--- a/tests/08-nat46.sh
+++ b/tests/08-nat46.sh
@@ -23,8 +23,8 @@ docker network inspect $TEST_NET 2> /dev/null || {
         docker network create --ipv6 --subnet ::1/112 --ipam-driver cilium --driver cilium $TEST_NET
 }
 
-docker run -dt -ti --net=$TEST_NET --name server -l $SERVER_LABEL $NETPERF_IMAGE
-docker run -dt -ti --net=$TEST_NET --name client -l $CLIENT_LABEL $NETPERF_IMAGE
+docker run -d -i --net=$TEST_NET --name server -l $SERVER_LABEL $NETPERF_IMAGE
+docker run -d -i --net=$TEST_NET --name client -l $CLIENT_LABEL $NETPERF_IMAGE
 
 CLIENT_IP=$(docker inspect --format '{{ .NetworkSettings.Networks.cilium.GlobalIPv6Address }}' client)
 CLIENT_IP4=$(docker inspect --format '{{ .NetworkSettings.Networks.cilium.IPAddress }}' client)

--- a/tests/08-nat46.sh
+++ b/tests/08-nat46.sh
@@ -54,6 +54,9 @@ cat <<EOF | cilium -D policy import -
 }
 EOF
 
+cilium endpoint config ${CLIENT_ID} NAT46=true
+cilium endpoint config ${SERVER_ID} NAT46=true
+
 function connectivity_test64() {
         # ICMPv4 echo request from client to server should succeed
         monitor_clear

--- a/tests/run-tests
+++ b/tests/run-tests
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+set -e
+
 for test in *.sh; do
 	./$test
 done

